manifests/backup/pg_dump.pp

```diff
@@ -0,0 +1,151 @@
+# @summary
+#   "Provider" for pg_dump backup
+#
+# @api private
+#
+# @param compress
+#   Whether or not to compress the backup. Support for compression also depends on other backup parameters.
+# @param databases
+#   Databases to backup. By default `[]` will back up all databases.
+# @param db_user
+#   PostgreSQL user to create with superuser privileges.
+# @param db_password
+#   Password to create for `$db_user`.
+# @param dir
+#   Directory to store backup.
+# @param dir_mode
+#   Permissions applied to the backup directory. This parameter is passed directly to the file resource.
+# @param dir_owner
+#   Owner for the backup directory. This parameter is passed directly to the file resource.
+# @param dir_group
+#   Group owner for the backup directory. This parameter is passed directly to the file resource.
+# @param format
+#   Backup format to use, must be supported by pg_dump or pg_dumpall. The choice will affect other options, i.e. compression.
+# @param install_cron
+#   Manage installation of cron package.
+# @param manage_user
+#   Manage creation of the backup user.
+# @param optional_args
+#   Specifies an array of optional arguments which should be passed through to the backup tool. These options are not validated, unsupported options may break the backup.
+# @param postscript
+#   One or more scripts that are executed when the backup is finished. This could be used to sync the backup to a central store.
+# @param prescript
+#   One or more scripts that are executed before the backup begins.
+# @param rotate
+#   Backup rotation interval in 24 hour periods.
+# @param success_file_path
+#   Specify a path where upon successful backup a file should be created for checking purposes.
+# @param time
+#   An array of two elements to set the backup time. Allows `['23', '5']` (i.e., 23:05) or `['3', '45']` (i.e., 03:45) for HH:MM times.
+# @param weekday
+#   Weekdays on which the backup job should run. Defaults to `*`. This parameter is passed directly to the cron resource.
+#
+class postgresql::backup::pg_dump (
+  Boolean $compress = true,
+  Array $databases = [],
+  Boolean $delete_before_dump = false,
+  String $dir,
+  String $dir_group = '0',
+  String $dir_mode = '0700',
+  String $dir_owner = 'root',
+  Enum['present','absent'] $ensure = 'present',
+  Enum['plain','custom','directory','tar'] $format = 'plain',
+  Boolean $install_cron = true,
+  Boolean $manage_user = false,
+  Array $optional_args = [],
+  String $pgpass_path = '/root/.pgpass',
+  Integer $rotate = 30,
+  String $script_path = '/usr/local/sbin/pg_dump.sh',
```

Comment:
```suggestion
  Stdlib::Absolutepath $script_path = '/usr/local/sbin/pg_dump.sh',
```
also I think sbin is deprecated. Arch Linux for example moved away a decade ago: https://lists.archlinux.org/pipermail/arch-dev-public/2012-March/022625.html
I suggest `/usr/local/bin`

Reply:
According to `man hiera(7)` (on Ubuntu and on CentOS) this directory is still very much supported and it serves the right purpose:

```
       /usr/local/sbin
              Locally installed programs for system administration.
```

Arch Linux isn't even listed as supported operating system for this module, so I would not focus on this Linux distribution for now.